mos_tests/nova/nova_test.py

```diff
@@ -601,3 +606,80 @@ def test_boot_instance_from_volume_bigger_than_flavor_size(self):
         # Check that instance is reachable
         ping = common_functions.ping_command(self.floating_ip.ip)
         self.assertTrue(ping, "Instance after creation is not reachable")
+
+
+class TestNovaDeferredDelete(TestBase):
+    """Nova Deferred Delete test cases"""
+
+    @classmethod
+    @pytest.fixture(scope='class', autouse=True)
```

Comment:
Actually `teardown` will be executed only after last test-case.
For e.g. we have following code:
```
class TestNovaDeferredDelete(TestBase):

    @classmethod
    @pytest.fixture(scope='class', autouse=True)
    def setup(cls):
        logger.debug('~~ SETUP ~~')

    @classmethod
    @pytest.yield_fixture(scope='class', autouse=True)
    def teardown(cls):
        yield
        logger.debug('~~ TEARDOWN ~~')

    def test_1(self):
        logger.debug('test_1')
    def test_2(self):
        logger.debug('test_2')
    def test_3(self):
        logger.debug('test_3')
```
And the console output will be:
```
# py.test --pdb -s -v -ra --env=akor_Test_Deployment_MOS_CI mos_tests/nova/nova_test.py::TestNovaDeferredDelete

mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_1
  2016-04-22 12:47:06,967 [DEBUG] mos_tests.nova.nova_test:617: ~~ SETUP ~~
  2016-04-22 12:47:09,429 [DEBUG] mos_tests.nova.nova_test:626: test_1
  PASSED
  2016-04-22 12:47:09,429 [INFO] mos_tests.conftest:133: refresh clients fixtures

mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_2
  2016-04-22 12:47:11,883 [DEBUG] mos_tests.nova.nova_test:629: test_2
  PASSED
  2016-04-22 12:47:11,884 [INFO] mos_tests.conftest:133: refresh clients fixtures

mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_3
  2016-04-22 12:47:14,346 [DEBUG] mos_tests.nova.nova_test:632: test_3
  PASSED
  2016-04-22 12:47:14,347 [DEBUG] mos_tests.nova.nova_test:623: ~~ TEARDOWN ~~
```